coordinator_test.go

```diff
@@ -88,15 +94,42 @@ func TestNewCoordinator(t *testing.T) {
 				{tableName: "B", parentTableName: "", referencedBy: []string{}},
 				{tableName: "C", parentTableName: "", referencedBy: []string{"B"}},
 			},
+			indexes: nil,
 			want: []*table{
 				{tableName: "A"},
 				{tableName: "B"},
 				{tableName: "C", referencedBy: []*table{{tableName: "B"}}},
 			},
 		},
+		{
+			desc: "Child table has an interleaved index",
+			schemas: []*tableSchema{
+				{tableName: "A", parentTableName: ""},
+				{tableName: "B", parentTableName: "A"},
+			},
+			indexes: []*indexSchema{
+				{indexName: "Bi", baseTableName: "B", parentTableName: "B"},
+			},
+			want: []*table{
+				{tableName: "A", childTables: []*table{{tableName: "B"}}},
+			},
+		},
+		{
+			desc: "Child table has a global (non-interleaved) index",
+			schemas: []*tableSchema{
+				{tableName: "A", parentTableName: ""},
+				{tableName: "B", parentTableName: "A"},
+			},
+			indexes: []*indexSchema{
+				{indexName: "Bi", baseTableName: "B", parentTableName: "B"},
+			},
+			want: []*table{
+				{tableName: "A", childTables: []*table{{tableName: "B"}}},
```

Comment:
Maybe it's better to check if `hasGlobalIndex` is set as expected in `compareTables` function?

Reply:
I agree with that it's an important aspect to test, I've updated test code including `compareTables` function. Also I noticed that I missed to set parentTableName as blank for a test.. fixed it as well.